include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdlops_v2r4r2.hpp

```diff
@@ -120,20 +123,58 @@ struct GridwiseGemm_bk0mk1_bk0nk1_mn_xdlops_v2r4r2
     static constexpr auto I6 = Number<6>{};
     static constexpr auto I7 = Number<7>{};
 
+    // Bytes per 32 lds bank: 32 * 4 bytes
+    static constexpr auto BankLength = Number<128>{};
+    static constexpr auto ElePerBank = Number<BankLength / sizeof(FloatAB)>{};
+
     // K1 should be Number<...>
     static constexpr auto K1 = Number<K1Value>{};
 
-    __host__ __device__ static constexpr index_t GetSharedMemoryNumberOfByte()
+    // M1 & M0
+    static constexpr auto M1PerBlock = Number<ElePerBank / K1Value>{};
+    static constexpr auto M0PerBlock = Number<MPerBlock / M1PerBlock>{};
+    static constexpr auto M1Padding  = I4;
+
+    // N1 & N0
+    static constexpr auto N1PerBlock = Number<ElePerBank / K1Value>{};
+    static constexpr auto N0PerBlock = Number<NPerBlock / M1PerBlock>{};
+    static constexpr auto N1Padding  = I4;
```

Comment:
It seems to me these 6 value determines the LDS layout? If so, instead of calculating them inside gridwise gemm, please passing them into gridwise GEMM as template parameters.

also, a type of line 140
```static constexpr auto N0PerBlock = Number<NPerBlock / N1PerBlock>{};```
instead of
```static constexpr auto N0PerBlock = Number<NPerBlock / M1PerBlock>{};```

Reply:
OK. thanks for finding out this typo.

Reply:
They are made to be template. Typo fixed.